common/src/main/kotlin/streams/utils/ProcedureUtils.kt

```diff
@@ -0,0 +1,71 @@
+package streams.utils
+
+import org.neo4j.collection.RawIterator
+import org.neo4j.dbms.api.DatabaseManagementService
+import org.neo4j.exceptions.UnsatisfiedDependencyException
+import org.neo4j.kernel.internal.GraphDatabaseAPI
+import org.neo4j.logging.Log
+import org.neo4j.logging.internal.LogService
+import streams.extensions.execute
+import java.lang.invoke.MethodHandles
+import java.lang.invoke.MethodType
+
+
+object ProcedureUtils {
+    @JvmStatic
+    private val raftMachineClass: Class<*>? = try {
+        Class.forName("com.neo4j.causalclustering.core.consensus.RaftMachine")
+    } catch (e: ClassNotFoundException) {
+        null
+    }
+
+    @JvmStatic
+    private val isLeaderMethodHandle = raftMachineClass?.let {
+        val lookup = MethodHandles.lookup()
+        lookup.findVirtual(it, "isLeader", MethodType.methodType(Boolean::class.java))
+            .asType(MethodType.methodType(Boolean::class.java, Any::class.java))
+    }
+
+    fun clusterMemberRole(db: GraphDatabaseAPI): String {
+        val fallback: (Exception?) -> String = { e: Exception? ->
```

Comment:
I'm not certain why we use reflection to resolve the `RaftMachine` and then fall back on the `dbms.cluster.role` procedure. The latter is the supported mechanism and the implementation calls through to `RaftMachine#isLeader` same as this.

Reply:
The fallback is for an _extrema ratio_ if something went wrong, in case of `UnsatisfiedDependencyException` so in case the class is not in there we manually manage it as normal case